src/openrct2/world/Sprite.cpp

```diff
@@ -928,106 +925,106 @@ uint16_t remove_floating_sprites()
     return removed;
 }
 
-/**
- * Determines whether it's worth tweening a sprite or not when frame smoothing is on.
- */
-static bool sprite_should_tween(SpriteBase* sprite)
+static bool IsValidEntity(SpriteBase* ent)
 {
-    switch (sprite->sprite_identifier)
-    {
-        case SpriteIdentifier::Peep:
-        case SpriteIdentifier::Vehicle:
-            return true;
-        case SpriteIdentifier::Misc:
-        case SpriteIdentifier::Litter:
-        case SpriteIdentifier::Null:
-            return false;
-    }
-    return false;
+    if (ent->sprite_identifier == SpriteIdentifier::Null)
+        return false;
+    if (ent->sprite_index == SPRITE_INDEX_NULL)
+        return false;
+    return true;
 }
```

Comment:
> Given that we only tween peeps and vehicles

Doesn't this change that claim?

Reply:
I don't understand the question

Reply:
This method is now returning true for `SpriteIdentifier::Misc` and `SpriteIdentifier::Litter`, unless those also have `ent->sprite_index == SPRITE_INDEX_NULL`, so isn't it now tweening those two?

Reply:
Those sprites are never in the list.

Reply:
Hm actually now that I think about it, the sprite may be removed but re-used as something else, so yeah it probably makes sense to be specific here.

Reply:
Refactored that part, it takes the element out of the list and replaces it will null to on removal to ensure it doesn't consider re-allocated sprites still as valid.